### ISSUE TITLE
Apply rules for `IfStatement` (e.g. `@typescript-eslint/no-unnecessary-condition`) to `SvelteIfBlock`

### ISSUE THREAD

**Issue by DePasqualeOrg** [sveltejs/eslint-plugin-svelte#1331]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.35.0

### What version of `eslint-plugin-svelte` are you using?

3.12.2

### What did you do?

I'd like to use the rules from the `strictTypeChecked` eslint config in my SvelteKit projects. I've created a minimal reproduction using `sv create` in which I've demonstrated the various adjustments I've had to make to get this config to work. However, even with these adjustments, there are still problems.

Adding this block to `eslint.config.js` allows me to use the strict typechecked rules in TypeScript and Svelte files:

```typeScript
	{
		files: ['**/*.ts', '**/*.svelte'],
		extends: [...ts.configs.strictTypeChecked],
		languageOptions: {
			parserOptions: {
				project: ['./tsconfig.json']
			}
		}
	},
```

However, then I get parsing errors in the Svelte files when I run eslint:

```
/<redacted>/svelte-eslint-strict-typechecked-demo/src/routes/+layout.svelte
  1:8  error  Parsing error: '>' expected

/<redacted>/svelte-eslint-strict-typechecked-demo/src/routes/+page.svelte
  1:8  error  Parsing error: '>' expected
```

Omitting the base config (the first item in the array) resolves the parsing errors: `...ts.configs.strictTypeChecked.slice(1)`

However, then I get inconsistent behavior in Svelte files:

```svelte
<script lang="ts">
	const emptyString = '';

	// This results in an error due to @typescript-eslint/no-unnecessary-condition
	if (emptyString) {
		console.log('test');
	}
</script>

<!-- This does not result in an error -->
{#if emptyString}
	<p>test</p>
{/if}
```

Is this a bug with eslint-plugin-svelte, or is a different configuration needed in order for the strict typechecked rules to be applied correctly?

### What did you expect to happen?

---

### What actually happened?

---

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/DePasqualeOrg/svelte-eslint-strict-typechecked-demo

**Comment by baseballyama** [comment]
Below config works well.

```diff
--- a/eslint.config.js
+++ b/eslint.config.js
@@ -26,15 +26,21 @@ export default ts.config(
                        'no-undef': 'off'
                }
        },
-       {
-               files: ['**/*.ts', '**/*.svelte'],
-               extends: [...ts.configs.strictTypeChecked.slice(1)], // Omitting the base config (first item) resolves the parsing errors
-               languageOptions: {
-                       parserOptions: {
-                               project: ['./tsconfig.json']
+       ...ts.configs.strictTypeChecked.map((c) => {
+               return {
+                       ...c,
+                       files: ['**/*.ts', '**/*.svelte'],
+                       languageOptions: {
+                               parserOptions: {
+                                       projectService: true,
+                               extraFileExtensions: ['.svelte'],
+                                       parser: ts.parser,
+                                       svelteConfig,
+                                       project: ['./tsconfig.json']
+                               }
                        }
                }
-       },
+       }),
        {
                files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
                languageOptions: {
```

**Comment by DePasqualeOrg** [comment]
@baseballyama, this does not resolve the issue with inconsistent behavior in Svelte files. The example I included above still has the same behavior: `emptyString` in the script part of `page.svelte` results in an error, but `emptyString` in the template part of the file does not. The rules are not being consistently applied to the template part of Svelte files.

**Comment by baseballyama** [comment]
The reason is that Svelte’s `{#if}` block is not an `IfStatement` but a `SvelteIfBlock` in AST.
Therefore, if `eslint-plugin-svelte` were to override all ESLint rules related to `IfStatement`, the problem would be solved. But that’s not realistic.
It would be nice if we could somehow use Proxy or similar techniques so that, when invoked from certain rules, it could behave as an `IfStatement`.
